tools/assets/extract/extase_oot64/dlist_resources.py

```diff
@@ -0,0 +1,1505 @@
+
+from ..extase import (
+    RESOURCE_PARSE_SUCCESS,
+    ResourceParseWaiting,
+    Resource,
+    File,
+)
+from ..extase.cdata_resources import (
+    CDataResource,
+    CDataExt_Array,
+    CDataExt_Struct,
+    CDataExt_Value,
+    CDataExtWriteContext,
+    INDENT,
+)
+
+
+BEST_EFFORT = True
+
+VERBOSE_ColorIndexedTexturesManager = False
+VERBOSE_BEST_EFFORT_TLUT_NO_REAL_USER = True
+
+
+class MtxResource(CDataResource):
+    braces_in_source = False
+
+    def write_mtx(resource, memory_context, v, wctx: CDataExtWriteContext):
+        assert isinstance(v, dict)
+        assert v.keys() == {"intPart", "fracPart"}
+        intPart = v["intPart"]
+        fracPart = v["fracPart"]
+
+        f = wctx.f
+        f.write(wctx.line_prefix)
+        f.write("gdSPDefMtx(\n")
+
+        for i in range(4):
+            if i != 0:
+                f.write(",\n")
+            f.write(wctx.line_prefix + INDENT)
+
+            for j in range(4):
+                # #define IPART(x) (((s32)((x) * 0x10000) >> 16) & 0xFFFF)
+                xi = intPart[j][i]
+                # #define FPART(x)  ((s32)((x) * 0x10000) & 0xFFFF)
+                xf = fracPart[j][i]
+                # Reconstruct the `(s32)((x) * 0x10000)` but as a u32
+                # (u32 since intPart and fracPart are u16 arrays)
+                # This works since `(s32)((x) * 0x10000)` in the IPART and FPART
+                # macros could be switched to `(u32)(s32)((x) * 0x10000)` without issue
+                u32_x_s15_16 = (xi << 16) | xf
+                # Cast to s32 (`(s32)(u32)(s32)((x) * 0x10000)` == `(s32)((x) * 0x10000)`)
+                s32_x_s15_16 = (
+                    u32_x_s15_16
+                    if u32_x_s15_16 < 0x8000_0000
+                    else u32_x_s15_16 - 0x1_0000_0000
+                )
+                x = s32_x_s15_16 / 0x10000
+
+                if j != 0:
+                    f.write(", ")
+                f.write(f"{x}f")
+        f.write("\n")
+
+        f.write(wctx.line_prefix)
+        f.write(")")
+
+        return True
+
+    cdata_ext = CDataExt_Struct(
+        (
+            ("intPart", CDataExt_Array(CDataExt_Array(CDataExt_Value.u16, 4), 4)),
+            ("fracPart", CDataExt_Array(CDataExt_Array(CDataExt_Value.u16, 4), 4)),
+        )
+    ).set_write(write_mtx)
+
+    def get_c_declaration_base(self):
+        return f"Mtx {self.symbol_name}"
+
+    def get_c_reference(self, resource_offset: int):
+        if resource_offset == 0:
+            return f"&{self.symbol_name}"
+        else:
+            raise ValueError
+
+
+class VtxArrayResource(CDataResource):
+    def write_elem(resource, memory_context, v, wctx: CDataExtWriteContext):
+        assert isinstance(v, dict)
+        wctx.f.write(wctx.line_prefix)
+        wctx.f.write(
+            f"VTX({v['x']:6}, {v['y']:6}, {v['z']:6}, "
+            f"{v['s']:#7X}, {v['t']:#7X}, "
+            f"{v['crnx']:#04X}, {v['cgny']:#04X}, {v['cbnz']:#04X}, {v['a']:#04X})"
```

Comment:
Currently this produces hex values like `0X`. I see you have addressed this elsewhere so could you do the same here.
(see fmt_hex_s in scene_rooms_resources.py which could maybe be made more global?)